activesupport/test/message_verifier_test.rb

```diff
@@ -16,18 +16,133 @@ def load(value)
   end
 
   def setup
-    @verifier = ActiveSupport::MessageVerifier.new("Hey, I'm a secret!")
+    @secret = "Hey, I'm a secret!"
+    @verifier = ActiveSupport::MessageVerifier.new(@secret)
+    @payload = 'data'
+    @options = { expires_at: Time.local(2022), for: 'test' }
+    @claims = ActiveSupport::Claims.new(payload: @payload, **@options)
+  end
+
+  def test_valid_message
+    header, claims, digest = @verifier.generate(@payload, @options).split(".")
+    assert_not @verifier.valid_message?(nil)
+    assert_not @verifier.valid_message?("")
+    assert_not @verifier.valid_message?("\xff") # invalid encoding
+    assert_not @verifier.valid_message?("#{header.reverse}.#{claims}.#{digest}")
+    assert_not @verifier.valid_message?("#{header}.#{claims}.#{digest.reverse}")
+    assert_not @verifier.valid_message?("#{header}.#{claims.reverse}.#{digest}")
+    assert_not @verifier.valid_message?("purejunk")
+    assert_not @verifier.valid_message?("..")
+    assert_not @verifier.valid_message?("pure.junk.data")
+  end
+
+  def test_simple_round_tripping
+    message = @verifier.generate(@payload, @options)
+    assert_equal @payload, @verifier.verified(message, for: 'test')
+    assert_equal @payload, @verifier.verify(message, for: 'test')
+  end
+
+  def raise_invalid_signature_when_value_is_nil
```

Comment:
Ah, I see. And I do agree with you that it doesn't make sense to sign a nil value when using the verifier directly. Could it be because of some of the cookie stuff?

Anyway if we want to change this we probably need to go through a deprecation cycle. Which might be a bit of a hassle instead of just keeping it around. Look at it from another point of view, if it doesn't make sense to sign a nil value, then people probably wouldn't do it. Then our deprecation cycle is for naught :smile: